src/features/registry/registry.ts

```diff
@@ -207,13 +216,124 @@ const resolveImageId = multiCacheMemoizee(
 	},
 );
 
+const resolveImageLocation = multiCacheMemoizee(
+	async (
+		application: string,
+		release: string | undefined,
+		service: string | undefined,
+		tx: Tx,
+	): Promise<string | undefined> => {
+		try {
+			const [image] = await api.resin.get({
+				resource: 'image',
+				passthrough: { req: permissions.root, tx },
+				options: {
+					$top: 1,
+					$select: ['is_stored_at__image_location'],
+					$filter: {
+						release_image: {
+							$any: {
+								$alias: 'ri',
+								$expr: {
+									ri: {
+										is_part_of__release: {
+											$any: {
+												$alias: 'ipor',
+												$expr: {
+													ipor: {
+														status: 'success' as const,
```

Comment:
you don't need this here since the API's pine clients doesn't type check model properties :)
```suggestion
														status: 'success',
```